packages/caliper-ethereum/lib/ethereum-connector.js

```diff
@@ -255,32 +251,41 @@ class EthereumConnector extends ConnectorBase {
             status.SetStatusSuccess();
         };
 
-        if (request.args) {
-            if (contractInfo.gas && contractInfo.gas[request.verb]) {
-                params.gas = contractInfo.gas[request.verb];
-            } else if (contractInfo.estimateGas) {
-                params.gas = 1000 + await contractInfo.contract.methods[request.verb](...request.args).estimateGas();
-            }
+        try {
+            if (request.args) {
+                if (contractInfo.gas && contractInfo.gas[request.verb]) {
+                    params.gas = contractInfo.gas[request.verb];
+                } else if (contractInfo.estimateGas) {
+                    params.gas = 1000 + await contractInfo.contract.methods[request.verb](...request.args).estimateGas();
+                }
 
-            try {
-                receipt = await contractInfo.contract.methods[request.verb](...request.args)[methodType](params);
-                onSuccess(receipt);
-            } catch (err) {
-                onFailure(err);
-            }
-        } else {
-            if (contractInfo.gas && contractInfo.gas[request.verb]) {
-                params.gas = contractInfo.gas[request.verb];
-            } else if (contractInfo.estimateGas) {
-                params.gas = 1000 + await contractInfo.contract.methods[request.verb].estimateGas(params);
-            }
+                if (request.sender) {
```

Comment:
Shouldn't this block also be present in the `!request.args` path? Because right now the custom sender can only be used with transactions that require some arguments. Or am I missing something?

Reply:
Hello @aklenik, it has been ages since I worked with this. Looking back at my project, you are probably right.

You can check in the following file how I used this technique (round-robin): https://github.com/Viserius/FabricBesuBenchmark/blob/master/caliper-workspace/workload/besunft-transfer.js

As you can see, I had it put in a local args variable, which might have caused the confusion of putting it in the request.args branch. Would you mind making the necessary adjustments?

Reply:
Thanks for the quick response! I'll think about the correct branching strategy, and push a new commit to your branch (and will ask @davidkel to review the PR, since I probably can't review my own contribution 😄)